tool/main.py

```diff
@@ -210,8 +242,13 @@ def get_deps(folder_path, project_repo_name, release_version, package_manager):
         deps_list_all = extract_deps.extract_deps_from_npm(npm_file)
 
     elif package_manager == "maven":
-        pom_xml_content, _, _ = get_lockfile(project_repo_name, release_version, package_manager)
-        deps_list_all = extract_deps.extract_deps_from_maven(pom_xml_content)
+        # Maven is more complex, because of child packages in the repo/pom; this requires to clone the whole repo
+        # TODO: Issue: not sure if this works with child projects -- probably not?
+        # Example: parent package A has a child package B; we want to run it on package B, but cloning won't work here (?)
```

Comment:
In the context of `maven`, I assume that you mean `submodule` instead of `package`.

The submodules could be dependent upon the build of their parent modules and that is why only cloning `module B` won't work.